PhysicsTools/TagAndProbe/interface/TriggerCandProducer.icc

```diff
@@ -337,11 +337,3 @@ void TriggerCandProducer<object>::beginRun(edm::Run const& iRun, edm::EventSetup
   if (printIndex_ && changed_)
     std::cout << "HLT configuration changed !" << std::endl;
 }
-
-template <class object>
-void TriggerCandProducer<object>::beginJob() {
-  filterName_ = "";
```

Comment:
Shouldn't `filterName_` get initialized anyhow somewhere (maybe in the ctor)?

Reply:
That isn't necessary. `std::string`'s default constructor already initializes itself to an empty string. In addition, if you do not explicitly declare the constructor call for a member data, the compiler will automatically use the classes default constructor (if the class has one). Therefore we are already guaranteed to have the member properly initialized (which is why I removed the line).